### PR TITLE
Extend Update/Delete queries with returning statement

### PR DESCRIPTION
**Pull request by Zhaars**
Extend Update/Delete queries with Returning

## Checklist

- [x] Change is covered by automated tests
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes
- [ ] CHANGELOG.md is updated (in case of notable or breaking changes)
- [ ] CHANGELOG_DEV.md is updated
- [ ] Benchmark results are attached (if applicable)
- [ ] [Example projects and code samples] are up-to-date (in case of API changes) 

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: encryptor/queryDataEncryptor.go [hunk 1]
```diff
@@ -326,8 +326,16 @@ func (encryptor *QueryDataEncryptor) encryptUpdateQuery(ctx context.Context, upd
 	if len(tables) == 0 {
 		return false, nil
 	}
+
 	qualifierMap := NewAliasToTableMapFromTables(tables)
 	firstTable := tables[0].TableName
+
+	// MySQL/MariaDB dont support returning after update statements
+	// Postgres doest but expect only one table in tables expression, so we can take the firstTable for returning matching
```
Comment (Lagovas) [inline encryptor/queryDataEncryptor.go:334]: actually, postgresql allows more than 1 table:
`An expression to be computed and returned by the UPDATE command after each row is updated. The expression can use any column names of the table named by table_name or table(s) listed in FROM. Write * to return all columns.`

File: encryptor/queryDataEncryptor.go [hunk 2]
```diff
@@ -398,6 +406,22 @@ func (encryptor *QueryDataEncryptor) onSelect(ctx context.Context, statement *sq
...
+	tables := GetTablesWithAliases(delete.TableExprs)
+	if !encryptor.hasTablesToEncrypt(tables) {
+		return false, nil
+	}
+	if len(tables) == 0 {
+		return false, nil
+	}
+
+	if encryptor.encryptor == nil {
+		return false, encryptor.onReturning(ctx, delete.Returning, tables[0].TableName.Name.ValueForConfig())
```
Comment (Lagovas) [inline encryptor/queryDataEncryptor.go:419]: same here, but allows to use tables from `DELETE ... USING`

**Review by Lagovas — APPROVED** [review]
in general looks cool

File: encryptor/queryDataEncryptor_test.go [hunk 3]
```diff
@@ -898,38 +907,184 @@ schemas:
...
 		}
+	})
 
-		for i := range returningColumns {
-			if _, ok := expectedNilColumns[i]; ok {
-				continue
+	t.Run("RETURNING columns with sql literals and several tables", func(t *testing.T) {
+		sqlparser.SetDefaultDialect(postgresql.NewPostgreSQLDialect())
+
+		returning := "specified_client_id_2, specified_unknown_column, default_client_id_2, default_unknown_column"
```
[the remark this answers is not in the document]
Comment (Zhaars) [reply encryptor/queryDataEncryptor_test.go:1040]: Sure, added

**Review by Lagovas — APPROVED** [review]

**Pull request merged**